python/ray/data/dataset.py

```diff
@@ -845,15 +846,115 @@ def groupby(self, key: "GroupKeyT") -> "GroupedDataset[T]":
         from ray.data.grouped_dataset import GroupedDataset
         return GroupedDataset(self, key)
 
-    def aggregate(self, *aggs: AggregateFn) -> U:
-        """Aggregate the entire dataset as one group.
+    def agg(self, aggs: Dict[AggregateOnT, List["Aggregation"]]):
```

Comment:
```suggestion
    def aggregate(self, aggs: Dict[AggregateOnT, List["Aggregation"]]):
```

I think the long form is preferable for clarity, as this isn't that common.

Reply:
Long-form is already added as an alias (see below this function definition), and the `.agg()` shorthand is the more commonly used one for Pandas users (Pandas has both `.agg()` and `.aggregate()` as well).

Reply:
I don't think we should have both.